File: src/1232-Check-If-It-Is-a-Straight-Line/1232.go
func checkStraightLine(c [][]int) bool {
    for i := 2; i < len(c); i++ {
        if (c[i][0] - c[0][0]) * (c[i][1] - c[1][1]) != (c[i][0] - c[1][0]) * (c[i][1] - c[0][1]) {
            return false
        }
    }
    return true
}

File: src/1232-Check-If-It-Is-a-Straight-Line/1232.cpp
class Solution 
{
public:
    bool checkStraightLine(vector<vector<int>>& c) 
    {
        for (int i = 2; i < c.size(); i++)
        {
            if ((c[i][0] - c[0][0]) * (c[i][1] - c[1][1]) != 
                (c[i][0] - c[1][0]) * (c[i][1] - c[0][1])) 
                return false;
        }
        return true;
    }
};